File: src/models/product_model.ts
import mongoose from 'mongoose';

export interface IProduct {
  productId: string;
  productName?: string | null | undefined;
  productDescription?: string | null | undefined;
  productPrice?: string | null | undefined;
  productCategory?: string | null | undefined;
  productImage?: string | null | undefined;
  productRating?: number | null | undefined; 
  createdAt?: Date; 
}

const productSchema = new mongoose.Schema<IProduct>({
  productId: { type: String, required: true },
  productName: { type: String },
  productDescription: { type: String },
  productPrice: { type: String },
  productCategory: { type: String },
  productImage: { type: String },
  productRating: { type: Number },
}, { timestamps: true }); // Enable timestamps

export const ProductModel = mongoose.model<IProduct>('Product', productSchema);

File: lib/models/product_model.dart
import 'package:json_annotation/json_annotation.dart';

part 'product_model.g.dart';

@JsonSerializable()
class Product {
  String? productId;
  String? productName;
  String? productDescription;
  String? productPrice;
  String? productCategory;
  String? productImage;
  double? productRating;
  Product({
    this.productId,
    this.productName,
    this.productDescription,
    this.productPrice,
    this.productCategory,
    this.productImage,
    this.productRating,
  });

  factory Product.fromJson(Map<String, dynamic> json) => _$ProductFromJson(json);
  Map<String, dynamic> toJson() => _$ProductToJson(this);
}